feat: add ability to Choose between different currencies when buying

satyam-mishra-pce opened the pull request:
1. Create a token config based on chains.
2. Create a new hook to handle all the calculations related to orders, and directly expose the computed values.
3. Create a new combobox component to be used for switching the currencies.
4. Update the entire payment flow to:
    - inform the user if a chain is not supported.
    - give an option to user to switch among the available and supported currencies on the selected chain.
    - display the wallet funds of the currency selected by the user.
    - display the metrics based on the preferred currency, rather than USD.
5. Created a new context for passing the hypercert data from server to client component, with bigint serialization.
![image](https://github.com/user-attachments/assets/714f37b6-cc07-4d11-ad5f-d3fd71ee8c48)

s-adamantine commented on app/hypercert/[hypercertId]/components/PaymentFlow/Prerequisites.tsx:
```diff
@@ -41,52 +43,63 @@ const PrerequisiteCard = ({
-	supportedChainId,
+	orderChainIdsSupportedByApp,
+	isCurrentChainSupportedByApp,
+	isCurrentChainSupportedByOrders,
```
@satyam_mishra13 what is the difference between `orderChainIdsSupportedByApp` and `isCurrentChainSupportedByApp`?

right now a hypercert can only exist in one chain at a time

satyam-mishra-pce replied:
`orderChainIdsSupportedByApp` is an array of supported chains. It has only one chain for now... but I've kept it the way so that it's almost no work when the app scales up to support more chains.

s-adamantine commented:
<img width="1208" alt="image" src="https://github.com/user-attachments/assets/1b94527a-d05c-4d43-a33d-5bbe7f305f09" />


works on sepolia, but not on mainnet. (this hypercert should show CELO instead of USD) http://localhost:3000/hypercert/42220-0x16bA53B74c234C870c61EFC04cD418B8f2865959-29944848289042584784776965453995602608128

satyam-mishra-pce commented:
> <img alt="image" width="1413" src="https://private-user-images.githubusercontent.com/31265876/409507130-ec506342-8740-4b3d-94c2-f7e2a2494e50.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg2NzIyODcsIm5iZiI6MTczODY3MTk4NywicGF0aCI6Ii8zMTI2NTg3Ni80MDk1MDcxMzAtZWM1MDYzNDItODc0MC00YjNkLTk0YzItZjdlMmEyNDk0ZTUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA0VDEyMjYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzNzIzNGE1NjM2Njg3NzEwMmRhMGVmNDY5NWJlNDc2ZGJlMGFkZjVlODdjMjgzYjIyMDcwODE5OWRlZGQ5YmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DrliIiCIA5ZzQr18Aw8uv4T4oc3sR9FOXFYRD4JE7Po">
> came across this error when trying to buy the hypercert
> 
> http://localhost:3000/hypercert/42220-0x16bA53B74c234C870c61EFC04cD418B8f2865959-29944848289042584784776965453995602608128

Thanks for the logs. I will look into this.

satyam-mishra-pce commented:
Just created the PR: https://github.com/GainForest/ecocertain/pull/41/commits/d1bb991e9c7ddc4256ca390224664b18efa93543

- This PR fixes the issue mentioned in https://github.com/GainForest/ecocertain/pull/41#issuecomment-2633462340
Also:
- fixed the slider lag while scrubbing it
- fixed bugs where the price calculation was causing errors.
- fixed bug where no funds in wallet was being displayed as "NaN"
